lib/pavilion/builder.py

```diff
@@ -608,6 +612,16 @@ def _setup_build_dir(self, dest):
                     "Could not copy extra file '{}' to dest '{}': {}"
                     .format(path, dest, err))
 
+    def make_files(self, files, dst):
+        for file in files:
```

Comment:
Files will be a dict of lists of strings: `{'foo/bar': ['line 1', 'line 2']}`. 
You can iterate over both the keys and values with:
```
for filename, content in files.items():
  ...
```